docs/rfc/grouping-api.md

```diff
@@ -0,0 +1,55 @@
+# Grouping API
+
+We will be adding the ability to add 'grouping' support to `react-beautiful-dnd`. I am attempting to create naming for the new api that feels right. The grouping feature will enable ability to hover over a Draggable to group with it. Right now I have been calling the feature 'grouping'. However, I am not sure if even that is the right name for the feature.
+
+What the consumer actually achieves with this feature may vary. By default when dropping a item over another one, the dragging item will drop _into_ the one being hovered over
+
+## API
+
+### `Draggable` > isGroupingEnabled
+
+This would allow a particular Draggable to be grouped with
+
+### `Draggable` > `DraggableStateSnapshot`
+
+```diff
+export type StateSnapshot = {|
+  isDragging: boolean,
+  draggingOver: ?DroppableId,
+-  isDropAnimating: boolean,
++  dropping: ?DroppingState,
++  groupingWith: ?DraggableId,
++  groupedOverBy: ?DraggableId,
+|};
+
++export type DroppingState = {|
++  reason: DropReason,
++  duration: number,
++|};
+```
+
+We have moved from `isDropAnimating` to `DroppingState` to give some more information about a drop. This is useful in crafting customised drop styles
+
+> I am not happy with the naming of these and I am looking for input!
```

Comment:
I am thinking of moving to the language of `combine` as I think it communicates the idea more accurately than 'grouping'

```diff
type StateSnapshot = {|
  isDragging: boolean,
  dropping: ?DroppingState,
  draggingOver: ?DroppableId,
+  combineWith: ?DraggableId,
+  combineTargetFor: ?DraggableId,
|};
```